pkg/token/service.go

```diff
@@ -127,10 +127,13 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 	// the cache is updated to ensure a cache hit from the first request.
 	// TODO: Maybe !idCfg.Init || idCfg.TokenFile.Use()
 	if !idCfg.Init || idCfg.TokenFile.AccessToken.Use || idCfg.TokenFile.RoleToken.Use {
-		// TODO: if cap(errs) == len(errs), implies all token updates failed, should be fatal
-		for _, err := range ts.updateTokenCachesAndWriteFiles(ctx, config.DEFAULT_MAX_ELAPSED_TIME_ON_INIT) {
+		errs := ts.updateTokenCachesAndWriteFiles(ctx, config.DEFAULT_MAX_ELAPSED_TIME_ON_INIT)
+		for _, err := range errs {
 			log.Errorf("Failed to refresh tokens after multiple retries: %s", err.Error())
 		}
+		if idCfg.Init && len(errs) != 0 {
+			return nil, fmt.Errorf("Unable to write token files for init mode: Failed to refresh tokens after multiple retries")
```

Comment:
[Fatal] Failed to ... load certificate daemon: k8s-athenz-sia deliberately fails to load if every token is not fetched during the init mode

Reply:
(dont include `k8s` because this is used for opestack too)